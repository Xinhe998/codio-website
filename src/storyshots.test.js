import initStoryshots from '@storybook/addon-storyshots';

// Runner
initStoryshots({
  storyKindRegex: /^((?!.*?Editors|DateInput).)*$/,
});
