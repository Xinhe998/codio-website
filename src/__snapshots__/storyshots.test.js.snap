// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots AppHeader 不顯示Tab 1`] = `
<header
  className="AppHeader"
>
  <div
    className="AppHeader__logo-wrap"
  >
    <img
      alt="codio_logo"
      className="codio_logo"
      src="codio_logo.png"
    />
  </div>
  <div
    className="AppHeader__btnGroup"
  >
    <svg
      className="shareBtn"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
      />
    </svg>
    <div
      className="userinfo"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(user-solid.svg)",
        }
      }
    >
      <img
        src="user-solid.svg"
      />
    </div>
  </div>
</header>
`;

exports[`Storyshots AppHeader 不顯示分享按鈕 1`] = `
<header
  className="AppHeader"
>
  <div
    className="AppHeader__logo-wrap"
  >
    <img
      alt="codio_logo"
      className="codio_logo"
      src="codio_logo.png"
    />
  </div>
  <div
    className="AppHeader__btnGroup"
  >
    <div
      className="userinfo"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(user-solid.svg)",
        }
      }
    >
      <img
        src="user-solid.svg"
      />
    </div>
  </div>
</header>
`;

exports[`Storyshots AppHeader 顯示Tab 1`] = `
<header
  className="AppHeader"
>
  <div
    className="AppHeader__logo-wrap"
  >
    <img
      alt="codio_logo"
      className="codio_logo"
      src="codio_logo.png"
    />
  </div>
  <ul
    className="AppHeader__tablist"
    role="tablist"
  >
    <li
      aria-disabled="false"
      aria-selected="true"
      className=""
      onClick={[Function]}
      role="tab"
      tabIndex="0"
    >
      HTML
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      className=""
      onClick={[Function]}
      role="tab"
      tabIndex="1"
    >
      CSS
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      className=""
      onClick={[Function]}
      role="tab"
      tabIndex="2"
    >
      JS
    </li>
    <li
      aria-disabled="false"
      aria-selected="false"
      className=""
      onClick={[Function]}
      role="tab"
      tabIndex="3"
    >
      Console
    </li>
  </ul>
  <div
    className="AppHeader__btnGroup"
  >
    <svg
      className="shareBtn"
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
      />
    </svg>
    <div
      className="userinfo"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(user-solid.svg)",
        }
      }
    >
      <img
        src="user-solid.svg"
      />
    </div>
  </div>
</header>
`;

exports[`Storyshots Button disabled 1`] = `
<button
  className="my_btn large type_primary"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button shape: square 1`] = `
<button
  className="my_btn large type_primary shape_square"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M337.509 305.372h-17.501l-6.571-5.486c20.791-25.232 33.922-57.054 33.922-93.257C347.358 127.632 283.896 64 205.135 64 127.452 64 64 127.632 64 206.629s63.452 142.628 142.225 142.628c35.011 0 67.831-13.167 92.991-34.008l6.561 5.487v17.551L415.18 448 448 415.086 337.509 305.372zm-131.284 0c-54.702 0-98.463-43.887-98.463-98.743 0-54.858 43.761-98.742 98.463-98.742 54.7 0 98.462 43.884 98.462 98.742 0 54.856-43.762 98.743-98.462 98.743z"
    />
  </svg>
  
</button>
`;

exports[`Storyshots Button size: large 1`] = `
<button
  className="my_btn large type_primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button theme: red 1`] = `
<button
  className="my_btn large type_primary theme_red"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button type: link 1`] = `
<button
  className="my_btn type_link"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button type: outline 1`] = `
<button
  className="my_btn small type_outline"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button type: primary 1`] = `
<button
  className="my_btn small type_primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Checkbox checked 1`] = `
<div>
  <label
    className="checkbox"
  >
    <input
      checked={true}
      name="myCheckbox"
      type="checkbox"
    />
    <span
      className="text"
    >
      
    </span>
    <span
      className="checkMark"
    />
  </label>
</div>
`;

exports[`Storyshots Checkbox default 1`] = `
<div>
  <label
    className="checkbox"
  >
    <input
      checked={false}
      name="myCheckbox"
      type="checkbox"
    />
    <span
      className="text"
    >
      
    </span>
    <span
      className="checkMark"
    />
  </label>
</div>
`;

exports[`Storyshots Modal default 1`] = `
<div
  className="Modal Modal__backdrop"
>
  <div
    className="Modal__wrapper"
  >
    <div
      className="Modal__title"
    >
      My PopUp Modal
      <svg
        className="close_btn"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </div>
    <div
      className="Modal__content"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </div>
  </div>
</div>
`;

exports[`Storyshots Modal 顯示control buttons 1`] = `
<div
  className="Modal Modal__backdrop"
>
  <div
    className="Modal__wrapper"
  >
    <div
      className="Modal__title"
    >
      My PopUp Modal
      <svg
        className="close_btn"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </div>
    <div
      className="Modal__content"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
      <div
        className="Modal__button-group"
      >
        <button
          className="Modal__button-group__cancel"
          onClick={[Function]}
          type="button"
        >
          取消
        </button>
        <button
          className="Modal__button-group__confirm"
          onClick={[Function]}
          type="button"
        >
          確定
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots MultiSelect default 1`] = `
<div>
  <div
    className="MultiSelect"
  >
    <span
      className="MultiSelect__title"
    >
      類別
    </span>
    <div
      className="MultiSelect__input-container"
      onClick={[Function]}
    >
      <input
        aria-busy="false"
        aria-expanded="false"
        aria-haspopup="true"
        aria-owns="MultiSelect__input-container"
        className="MultiSelect__input"
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        role="listbox"
        size={2}
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots ProjectCard default 1`] = `
<div
  className="ProjectCard"
>
  <div
    className="ProjectCard__container"
  >
    <img
      alt="projectThumbImgage"
      className="ProjectCard__thumbImg"
      src="smartmockups_jxfuqv8i.jpg"
    />
    <div
      className="ProjectCard__info"
    >
      <img
        alt="avatar"
        className="ProjectCard__avatar"
        src="https://avatars3.githubusercontent.com/u/29070256?s=460&v=4"
      />
      <div
        className=""
      >
        <p
          className="ProjectCard__title"
        >
          todolist
        </p>
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M255.8 112c-80.4 0-143.8 50.6-219.6 133.3-5.5 6.1-5.6 15.2-.1 21.3C101 338.3 158.2 400 255.8 400c96.4 0 168.7-77.7 220.1-134 5.3-5.8 5.6-14.6.5-20.7C424 181.8 351.5 112 255.8 112zm4.4 233.9c-53 2.4-96.6-41.2-94.1-94.1 2.1-46.2 39.5-83.6 85.7-85.7 53-2.4 96.6 41.2 94.1 94.1-2.1 46.2-39.5 83.6-85.7 85.7z"
            />
            <path
              d="M256 209c0-6 1.1-11.7 3.1-16.9-1 0-2-.1-3.1-.1-36.9 0-66.6 31.4-63.8 68.9 2.4 31.3 27.6 56.5 58.9 58.9 37.5 2.8 68.9-26.9 68.9-63.8 0-1.3-.1-2.6-.1-3.9-5.6 2.5-11.7 3.9-18.2 3.9-25.2 0-45.7-21.1-45.7-47z"
            />
          </svg>
          1300
        </span>
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M352 56h-1c-39.7 0-74.8 21-95 52-20.2-31-55.3-52-95-52h-1c-61.9.6-112 50.9-112 113 0 37 16.2 89.5 47.8 132.7C156 384 256 456 256 456s100-72 160.2-154.3C447.8 258.5 464 206 464 169c0-62.1-50.1-112.4-112-113z"
            />
          </svg>
          560
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots ProjectCard 顯示名次 1`] = `
<div
  className="ProjectCard"
>
  <div
    className="ProjectCard__place"
  >
    1
  </div>
  <div
    className="ProjectCard__container"
  >
    <img
      alt="projectThumbImgage"
      className="ProjectCard__thumbImg"
      src="smartmockups_jxfuqv8i.jpg"
    />
    <div
      className="ProjectCard__info"
    >
      <img
        alt="avatar"
        className="ProjectCard__avatar"
        src="https://avatars3.githubusercontent.com/u/29070256?s=460&v=4"
      />
      <div
        className=""
      >
        <p
          className="ProjectCard__title"
        >
          todolist
        </p>
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M255.8 112c-80.4 0-143.8 50.6-219.6 133.3-5.5 6.1-5.6 15.2-.1 21.3C101 338.3 158.2 400 255.8 400c96.4 0 168.7-77.7 220.1-134 5.3-5.8 5.6-14.6.5-20.7C424 181.8 351.5 112 255.8 112zm4.4 233.9c-53 2.4-96.6-41.2-94.1-94.1 2.1-46.2 39.5-83.6 85.7-85.7 53-2.4 96.6 41.2 94.1 94.1-2.1 46.2-39.5 83.6-85.7 85.7z"
            />
            <path
              d="M256 209c0-6 1.1-11.7 3.1-16.9-1 0-2-.1-3.1-.1-36.9 0-66.6 31.4-63.8 68.9 2.4 31.3 27.6 56.5 58.9 58.9 37.5 2.8 68.9-26.9 68.9-63.8 0-1.3-.1-2.6-.1-3.9-5.6 2.5-11.7 3.9-18.2 3.9-25.2 0-45.7-21.1-45.7-47z"
            />
          </svg>
          1300
        </span>
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M352 56h-1c-39.7 0-74.8 21-95 52-20.2-31-55.3-52-95-52h-1c-61.9.6-112 50.9-112 113 0 37 16.2 89.5 47.8 132.7C156 384 256 456 256 456s100-72 160.2-154.3C447.8 258.5 464 206 464 169c0-62.1-50.1-112.4-112-113z"
            />
          </svg>
          560
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots RadioButtonGroup controlled 1`] = `
<div>
  <div
    className="radioButtonGroup required"
  >
    <span
      className="radioButtonGroup__title"
    >
      Radio Button Group
    </span>
    <div
      className="radioButtonGroup__container"
    >
      <label
        className="radioButtonGroup__option checked"
      >
        Option 1
        <input
          checked={true}
          id="Option 1"
          name="my_radio_group"
          onChange={[Function]}
          type="radio"
          value="Option 1"
        />
        <span
          className="checkmark"
        />
      </label>
      <label
        className="radioButtonGroup__option"
      >
        Option 2
        <input
          checked={false}
          id="Option 2"
          name="my_radio_group"
          onChange={[Function]}
          type="radio"
          value="Option 2"
        />
        <span
          className="checkmark"
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`Storyshots RadioButtonGroup default 1`] = `
<div
  className="radioButtonGroup required"
>
  <span
    className="radioButtonGroup__title"
  >
    Radio Button Group
  </span>
  <div
    className="radioButtonGroup__container"
  >
    <label
      className="radioButtonGroup__option checked"
    >
      Option 1
      <input
        checked={true}
        id="Option 1"
        name="my_radio_group"
        onChange={[Function]}
        type="radio"
        value="Option 1"
      />
      <span
        className="checkmark"
      />
    </label>
    <label
      className="radioButtonGroup__option"
    >
      Option 2
      <input
        checked={false}
        id="Option 2"
        name="my_radio_group"
        onChange={[Function]}
        type="radio"
        value="Option 2"
      />
      <span
        className="checkmark"
      />
    </label>
  </div>
</div>
`;

exports[`Storyshots Resizer default 1`] = `
<div
  className="resizer"
  style={
    Object {
      "width": "undefinedpx",
    }
  }
>
  <div>
    <div
      style={
        Object {
          "backgroundColor": "#f0f0f0",
          "height": "150px",
        }
      }
    >
      Resize Me
    </div>
  </div>
  <div
    className="handle"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "left": "calc(undefinedpx - 9px)",
      }
    }
  />
</div>
`;

exports[`Storyshots SearchAlert default 1`] = `
<div
  className="SearchAlert"
>
  This is Search Alert
  <svg
    className="SearchAlert__CloseBtn"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
    />
  </svg>
</div>
`;

exports[`Storyshots TabPanel default 1`] = `
Array [
  <div
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    role="tabpanel"
  >
    Tab 1 Content
  </div>,
  <div
    className="react-tabs__tab-panel"
    role="tabpanel"
  />,
]
`;

exports[`Storyshots TextInput controlled 1`] = `
<div>
  <div
    className="textinput required"
  >
    <span
      className="textinput__title"
    >
      E-mail
    </span>
    <input
      disabled={false}
      onBlur={null}
      onChange={[Function]}
      onFocus={null}
      placeholder=""
      readOnly={false}
      type="text"
      value={Array []}
    />
    <div
      className="textinput__icon"
    />
  </div>
</div>
`;

exports[`Storyshots TextInput default 1`] = `
<div
  className="textinput required"
>
  <span
    className="textinput__title"
  >
    E-mail
  </span>
  <input
    disabled={false}
    onBlur={null}
    onFocus={null}
    placeholder=""
    readOnly={false}
    type="text"
    value=""
  />
  <div
    className="textinput__icon"
  />
</div>
`;
